python/ray/util/collective/collective_group/nccl_collective_group.py

```diff
@@ -406,25 +449,44 @@ def _get_nccl_collective_communicator(self, comm_key, device_list):
         actual_world_size = len(device_list) * self.world_size
         comms = [None] * len(device_list)
         streams = [None] * len(device_list)
+        events = [None] * len(device_list)
         nccl_util.groupStart()
         for i, device in enumerate(device_list):
             actual_rank = self.rank * len(device_list) + i
             with nccl_util.Device(device):
                 comms[i] = nccl_util.create_nccl_communicator(
                     actual_world_size, nccl_uid, actual_rank)
-                streams[i] = cupy.cuda.Stream.null
-                # Stream(non_blocking=True)
+                if _MULTI_STREAM:
+                    pool: StreamPool = self._dev_pool_map.get(comm_key, StreamPool(self.POOL_SIZE))
```

Comment:
every time when you try to query `_dev_pool_map` with `comm_key`, it is nonexistent, so you create a new Stream Pool? This is incorrect...